### ISSUE TITLE
Cataclysm: DDA build failure (`cataclysm-dda-0.F-3`) - `error: invalid conversion from 'const char*' to 'char*'`

### ISSUE THREAD

**Issue by mitchmindtree** [NixOS/nixpkgs#192002]
### Steps To Reproduce

1. build `cataclysm-dda` using `nixos-unstable` branch.

Here is the lock for the `nixpkgs` input I'm using:

```json
    "nixpkgs": {
      "locked": {
        "lastModified": 1663494472,
        "narHash": "sha256-fSowlaoXXWcAM8m9wA6u+eTJJtvruYHMA+Lb/tFi/qM=",
        "owner": "NixOS",
        "repo": "nixpkgs",
        "rev": "f677051b8dc0b5e2a9348941c99eea8c4b0ff28f",
        "type": "github"
      },
      "original": {
        "owner": "NixOS",
        "ref": "nixos-unstable",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
```

### Build log

```
$ nixos-rebuild switch --use-remote-sudo --flake .#desktop                                                                                       
warning: Git tree '/home/mindtree/programming/nix/mindtrix' is dirty                                                                                                                           
building the system configuration...                                                                                                                                                           
warning: Git tree '/home/mindtree/programming/nix/mindtrix' is dirty                                                                                                                           
error: builder for '/nix/store/1gr5viiidagbypayh993mn6s33a8psvf-cataclysm-dda-0.F-3.drv' failed with exit code 2;                                                                              
       last 10 log lines:                                                                                                                                                                      
       >       |                                            const char*                                                                                                                        
       > src/sdl_font.cpp:32:64: error: invalid conversion from 'const char*' to 'char*' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-fpermissive-fpermissive8;;]         
       >    32 |                     if( nullptr != ( ts = TTF_FontFaceStyleName( tf.get() ) ) ) {                                                                                             
       >       |                                           ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~                                                                                                   
       >       |                                                                |                                                                                                              
       >       |                                                                const char*                                                                                                    
       > At global scope:                                                                                                                                                                      
       > cc1plus: note: unrecognized command-line option '-Wno-unknown-warning-option' may have been intended to silence earlier diagnostics                                                   
       > make: *** [Makefile:962: obj/tiles/sdl_font.o] Error 1                                                                                                                                
       > make: *** Waiting for unfinished jobs....                                                                                                                                             
       For full logs, run 'nix log /nix/store/1gr5viiidagbypayh993mn6s33a8psvf-cataclysm-dda-0.F-3.drv'.                                                                                       
error: 1 dependencies of derivation '/nix/store/gcrq4pkxmyz6lz3xdck4yfc1zag8b39v-home-manager-path.drv' failed to build                                                                        
error: 1 dependencies of derivation '/nix/store/m2zmkid5sc6v96b7nvvqfwqr7wl7wznv-home-manager-generation.drv' failed to build                                                                  
error: 1 dependencies of derivation '/nix/store/hql0hppjj451060jmj58wq022k5zzi5a-user-environment.drv' failed to build                                                                         
error: 1 dependencies of derivation '/nix/store/jsw84lbwflrdv9dw30i2d9apf74f5rr1-etc.drv' failed to build                                                                                      
error: 1 dependencies of derivation '/nix/store/5cldvfrgv6cm4bm5p8f08aa88n4mi79i-nixos-system-minddesk-22.11.20220918.f677051.drv' failed to build
```

Find the full output here: https://gist.github.com/mitchmindtree/5cd6e2c4a2ef50ea6fa44744662b0327

### Notify maintainers

cc @mnacamura @DeeUnderscore 

### Metadata

```console
[user@system:~]$ nix run nixpkgs#nix-info -- -m
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.59, NixOS, 22.11 (Raccoon), 22.11.20220813.e105167`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.10.3`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by DeeUnderscore** [comment]
Should be solved by #191096 (which hasn't made it to nixos-unstable as of right now, but is already in nixpkgs-unstable)

**Comment by DeeUnderscore** [comment]
Closing this, since it's now working in both current nixpkgs-unstable and current nixos-unstable.
